### PR TITLE
docs(FAQ): add an entry on blocked cross-origin request

### LINKED ISSUE #1978: CORS error with `cors_enable=True` enabled
I have a website running at `https://domain.com` and an falcon API server running at `https://sub.domain.com`.

I want my website to be able to query the API server. That's why I have enabled `cors_enable=True` in my falcon app:

```python
app = falcon.App(middleware=[AuthMiddleware()], cors_enable=True)
```

Then, the client emits an axios request:

```js
axios({
  method: "post",
  url: "https://sub.domain.com",
  data: {
    data=someData,
  },
  headers: { Authorization: `Bearer ${idToken}` },
})
```

But I get a CORS error (using Firefox)

```
OPTIONS https://sub.domain.com CORS Missing Allow Origin

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://sub.domain.com  (Reason: CORS header 'Access-Control-Allow-Origin' missing).

Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://sub.domain.com (Reason: CORS request did not succeed)
```

However, when I use `httpie` on my laptop, the request works:

```sh
http POST https://sub.domain.com  data=someData
```

What am I doing wrong?

### PR DESCRIPTION
**Pull request by vytas7**
This PR extends our FAQ with the answers that I provided on #1978.

Closes #1978

### CODE REVIEW

File: docs/user/faq.rst [hunk 1]
```diff
@@ -160,6 +160,56 @@ try one of the generic
...
+            # NOTE: Do not authenticate OPTIONS requests.
+            if req.method == 'OPTIONS':
+                return
+
+            # -- snip --
+
+            # My authorization logic...
+
+Alternatively, if you rather prefer not to fiddle with third party middleware,
+you can subclass it:
```
Comment (CaselIT) [inline docs/user/faq.rst:196]: as is it does not seem clear. How about

> Alternatively, if you use a third party middleware that does not offer that options,
you can subclass it:

**Review by CaselIT — COMMENTED** [review]
look fine! just a line that can be improved

File: docs/user/faq.rst (hunk 1 above)
Comment (vytas7) [reply docs/user/faq.rst:196]: I agree my sentence isn't very clear... 

However, third party middleware not offering that option sounds somewhat strange too. Unless it is written in C and delivered compiled, you can well edit its code too :thinking:

Comment (CaselIT) [reply docs/user/faq.rst:196]: it's not usually practical to edit a library. Maybe I don't understand what you mentioned

Comment (vytas7) [reply docs/user/faq.rst:196]: > it's not usually practical to edit a library.

That's exactly what I was referring to. I'll try to reword to reflect just that.

**Review by CaselIT — APPROVED** [review]
looks clear now

**Pull request merged**